paperqa/types.py

```diff
@@ -80,7 +80,34 @@ def matches_filter_criteria(self, filter_criteria: dict) -> bool:
 class Text(Embeddable):
     text: str
     name: str
-    doc: Doc
+    # TODO: doc is often used as `DocDetails`.
+    # However, typing it as `Doc | DocDetails` makes SearchIndex.query
+    # to fail to return the correct results.
+    # Will keep it as `Doc` and cast to `DocDetails` when needed for now.
+    doc: Doc | DocDetails
+
+    @model_validator(mode="before")
+    @classmethod
+    def ensure_doc(cls, values: Any) -> Any:
+        if not isinstance(values, dict):
+            # NOTE: If I don't know what is being passed, I don't change it
+            # This happens in some tests
+            return values
+
+        doc_data = values.get("doc")
+        if isinstance(doc_data, Doc | DocDetails):
```

Comment:
```suggestion
        if isinstance(doc_data, Doc):
```

Since `DocDetails` is a `Doc`